### DISCUSSION TITLE
Hosted in a subdirectory without Docker

### DISCUSSION THREAD

**Discussion by TCB13** [FreshRSS/FreshRSS#5631] [Q&A] [answered]
The README.md is missing instructions for "Hosted in a subdirectory" (nginx) without Docker. Thank you.

**Comment by TCB13** [comment, marked as answer]
This works for me:

````
	# FreshRSS
	location /freshrss {
		alias /mnt/web/freshrss/p;
		index index.html index.htm index.php;
		try_files $uri $uri/ index.php;
		location ~ ^.+?\.php(/.*)?$ {
			include /etc/nginx/fastcgi_params;
	                fastcgi_pass unix:/run/php/php7.4-fpm.sock;
			fastcgi_split_path_info ^(.+\.php)(/.*)$;
			set $path_info $fastcgi_path_info;
			fastcgi_param PATH_INFO $path_info;
			fastcgi_param SCRIPT_FILENAME $request_filename;
		}
	}
````

`fastcgi_param SCRIPT_FILENAME $request_filename;` is very important here.

**Reply by Alkarex** [reply]
Check https://freshrss.github.io/FreshRSS/en/admins/10_ServerConfig.html

**Reply by TCB13** [reply]
That link only provides a configuration for a **subdomain**, I was trying to do it as a folder / **subdirectory**. Subdirectory requires a different `fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;` otherwise it won't work.

**Comment by Alkarex** [comment]
Hello,
Indeed, our documentation seems to have lost this information along the way (there is still a bit [in the French version ](https://github.com/FreshRSS/FreshRSS/blob/edge/README.fr.md#exemple-dinstallation-compl%C3%A8te-sur-linux-debianubuntu)). PR welcome.

Assuming Apache with default support for `.htaccess`, unpacking FreshRSS in a sub-folder (for instance at `https://example.net/freshrss/`) should be sufficient to get going.

For other Web servers or if support for `.htaccess` is disabled, you need in particular to make sure that the `./freshrss/data/` folder is denied to the public. Ideally, only the `./freshrss/p/` folder should be publicly available. Check e.g.:
* https://github.com/FreshRSS/FreshRSS/blob/edge/data/.htaccess
* https://github.com/FreshRSS/FreshRSS/blob/edge/p/.htaccess

**Reply by TCB13** [reply]
@Alkarex I'm sorry, I'm using nginx. What your said are all valid points even for nginx, however without the location block configured as described it wont work.
